pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -204,10 +204,12 @@ func toRecipeConfigDatamodel(config *RecipeConfigProperties) datamodel.RecipeCon
 				}
 			}
 
-			recipeConfig.Terraform.Providers = toRecipeConfigTerraformProvidersDatamodel(config)
+			recipeTFProviders := toRecipeConfigTerraformProvidersDatamodel(config)
+			recipeConfig.Terraform.Providers = recipeTFProviders
```

Comment:
```suggestion
			recipeConfig.Terraform.Providers = toRecipeConfigTerraformProvidersDatamodel(config)
```

Is there a reason why we don't do it like above?

Reply:
i was in middle of updating files. updated now.